cmd/new.go

```diff
@@ -112,12 +116,165 @@ func scaffoldRule(params newRuleCommandParams) error {
 	}
 
 	if params.type_ == "builtin" {
+		if err := addToDataYAML(params); err != nil {
+			return err
+		}
+
+		if err := addRuleToREADME(params); err != nil {
+			return err
+		}
+
 		return scaffoldBuiltinRule(params)
 	}
 
 	return fmt.Errorf("unsupported type %v", params.type_)
 }
 
+func addToDataYAML(params newRuleCommandParams) error {
+	// Read the YAML content from the file
+	yamlContent, err := os.ReadFile("bundle/regal/config/provided/data.yaml")
```

Comment:
This assumes the command is run from the project root directory. I guess that's an OK assumption to make, but perhaps we should check for that and fail with a message saying that `regal new rule --type builtin` must be run from that location?